lib/code_corps/model/project.ex

```diff
@@ -86,22 +85,13 @@ defmodule CodeCorps.Project do
   end
 
   defp associate_categories(changeset, %{"categories_ids" => []}) do
-    do_associate_categories(changeset, [])
+    changeset |> put_assoc(:categories, [])
   end
   defp associate_categories(changeset, %{"categories_ids" => ids}) do
-    categories =
-      ids
-      |> Enum.join(",")
-      |> coalesce_id_string()
-      |> find_categories()
-
+    categories = ids |> Enum.map(&String.to_integer/1) |> find_categories()
     changeset |> put_assoc(:categories, categories)
   end
   defp associate_categories(changeset, _) do
-    do_associate_categories(changeset, [])
-  end
-
-  defp do_associate_categories(changeset, _) do
     changeset |> put_assoc(:categories, [])
   end
```

Comment:
I'm reasonably sure you could still rewrite this

```Elixir
defp associate_categories(changeset, %{"categories_ids" => []}) do
  changeset |> put_assoc(:categories, [])
end
defp associate_categories(changeset, %{"categories_ids" => ids}) do
  categories = ids |> Enum.map(&String.to_integer/1) |> find_categories()
  changeset |> put_assoc(:categories, categories)
end
defp associate_categories(changeset, _) do
  changeset |> put_assoc(:categories, [])
end
```

as this

```Elixir
defp associate_categories(changeset, %{"categories_ids" => ids}) when is_list(ids) when length(ids) > 0 do
  categories = ids |> Enum.map(&String.to_integer/1) |> find_categories()
  changeset |> put_assoc(:categories, categories)
end
defp associate_categories(changeset, _) do
  changeset |> put_assoc(:categories, [])
end
```

Heck, since you're just using an enum map, you don't even need the `when length(ids) > 0` part.